src/java.base/share/classes/java/net/MulticastSocket.java

```diff
@@ -30,10 +30,14 @@
 import java.nio.channels.MulticastChannel;
 
 /**
- * The multicast datagram socket class is useful for sending
- * and receiving IP multicast packets. A MulticastSocket is
- * a (UDP) DatagramSocket, with additional capabilities for
- * joining "groups" of other multicast hosts on the internet.
+ * A {@code MulticastSocket} is a datagram socket that is
+ * convenient for sending and receiving IP multicast datagrams.
+ * The {@code MulticastSocket} constructors create a socket
+ * with appropriate socket options enabled that make
+ * it suitable for receiving multicast datagrams.
+ * The {@code MulticastSocket} class additionally defines
+ * convenient setter and getter methods for socket options
+ * that are commonly used by multicasting applications.
  * <P>
  * A multicast group is specified by a class D IP address
```

Comment:
I wonder does it require a line to introduce groups in a similar to the previous iteration of the doc?

Reply:
Hmmm... That's a good point. I added the suggested text, then noticed some inconsistencies in the example code that I had to fix. Now this will require a CSR I guess.